pyresample/geometry.py

```diff
@@ -2010,20 +2030,36 @@ def geocentric_resolution(self, ellps='WGS84', radius=None):
             dst = Proj("+proj=cart +a={} +b={}".format(radius, radius))
         else:
             dst = Proj("+proj=cart +ellps={}".format(ellps))
+        # need some altitude, go with the surface (0)
         alt_x = np.zeros(x.size)
         alt_y = np.zeros(y.size)
+        # convert our midlines to (X, Y, Z) geocentric coordinates
         hor_xyz = np.stack(transform(src, dst, x, mid_row_y, alt_x), axis=1)
         vert_xyz = np.stack(transform(src, dst, mid_col_x, y, alt_y), axis=1)
+        # Find the distance in meters along our midlines
         hor_dist = np.linalg.norm(np.diff(hor_xyz, axis=0), axis=1)
         vert_dist = np.linalg.norm(np.diff(vert_xyz, axis=0), axis=1)
-        dist = np.concatenate((hor_dist, vert_dist))
-        dist = dist[np.isfinite(dist)]
-        if not dist.size:
-            raise RuntimeError("Could not calculate geocentric resolution")
-        # return np.max(dist)  # alternative to histogram
+        # Get rid of any NaNs or infinite values
+        hor_dist = hor_dist[np.isfinite(hor_dist)]
+        vert_dist = vert_dist[np.isfinite(vert_dist)]
         # use the average of the largest histogram bin to avoid
-        # outliers and really large values
-        return np.mean(np.histogram_bin_edges(dist)[:2])
+        # outliers and really large values.
+        # Very useful near edge of disk geostationary areas.
```

Comment:
The large values near the disk edges would be very *helpful*. But I guess that would require to inspect the intersection of source and target areas so that cases where "too high" radii are not desired aren't affected.